### ISSUE TITLE
Uncaught Error: EventDispatcher is disposed

### ISSUE THREAD

**Issue by lticadler** [amcharts/amcharts4#3051]
**Bug description**

I get sometimes unfortunately not reproducible errors in my console after recreating charts.
Seems like some pending events try to do something.

1. Bar chart with 2 series.
2. hover to display tooltips
3. dispose chart
4. create new chart
5. hover again over the new chart

Mostly no problem but too often to ignore I get the error "Uncaught Error: EventDispatcher is disposed"

No pen :( because I don't have any idea how to...

Hopefully the stacks below provide you a hint.

**Environment (if applicable)**

 - amCharts version 4.10.6 (and earlier)
 - Browser and its version: Only detected yet in chrome (because the most used/tested browser)

**Stacks**
```
EventDispatcher.ts:369 Uncaught Error: EventDispatcher is disposed
    at t._on (EventDispatcher.ts:369)
    at t.on (EventDispatcher.ts:444)
    at e.setDataItem (Label.ts:1719)
    at e.set [as dataItem] (Sprite.ts:4203)
    at e.updateLegendValue (Series.ts:1375)
    at e.updateLegendValue (ColumnSeries.ts:1154)
    at e.showTooltipAtDataItem (XYSeries.ts:2002)
    at DateAxis.ts:2627
    at Object.e.each (Array.ts:95)
    at t.each (List.ts:915)
    
    	t._on	@	EventDispatcher.ts:369
t.on	@	EventDispatcher.ts:444
e.setDataItem	@	Label.ts:1719
set	@	Sprite.ts:4203
e.updateLegendValue	@	Series.ts:1375
e.updateLegendValue	@	ColumnSeries.ts:1154
e.showTooltipAtDataItem	@	XYSeries.ts:2002
(anonymous)	@	DateAxis.ts:2627
e.each	@	Array.ts:95
t.each	@	List.ts:915
e.showTooltipAtPosition	@	DateAxis.ts:2624
(anonymous)	@	XYChart.ts:1538
(anonymous)	@	Iterator.ts:100
(anonymous)	@	Iterator.ts:45
Wglt.e.each	@	Iterator.ts:99
e.showAxisTooltip	@	XYChart.ts:1535
e.handleCursorPositionChange	@	XYChart.ts:1227
(anonymous)	@	EventDispatcher.ts:444
(anonymous)	@	EventDispatcher.ts:346
e.d	@	AsyncPending.ts:162
t.update	@	System.ts:351
(anonymous)	@	System.ts:405
requestAnimationFrame (async)		
u	@	AsyncPending.ts:40
t.requestFrame	@	System.ts:404
e.invalidatePosition	@	Sprite.ts:1215
set	@	Sprite.ts:3660
(anonymous)	@	Preloader.ts:251
(anonymous)	@	EventDispatcher.ts:489
(anonymous)	@	EventDispatcher.ts:324
e.each	@	Array.ts:95
t._eachListener	@	EventDispatcher.ts:294
t.dispatchImmediately	@	EventDispatcher.ts:322
t.dispatchImmediately	@	Registry.ts:313
t.update	@	System.ts:125
(anonymous)	@	System.ts:405
requestAnimationFrame (async)		
u	@	AsyncPending.ts:40
t.requestFrame	@	System.ts:404
e.invalidateDataRange	@	Component.ts:983
e.invalidateProcessedData	@	Component.ts:2113
e.handleDataItemWorkingValueChange	@	Series.ts:1093
e.handleDataItemWorkingValueChange	@	XYSeries.ts:2657
e.setWorkingValue	@	DataItem.ts:832
e.setValue	@	DataItem.ts:748
e.setDate	@	DataItem.ts:950
set	@	XYSeries.ts:179
(anonymous)	@	Component.ts:726
(anonymous)	@	Object.ts:120
l	@	Object.ts:105
Qkdp.e.each	@	Object.ts:119
e.processDataItem	@	Component.ts:689
e.processDataItem	@	XYSeries.ts:1130
(anonymous)	@	Component.ts:1204
e.each	@	Array.ts:95
t.each	@	List.ts:915
o	@	Component.ts:1201
e.validateData	@	Component.ts:1194
e.validateData	@	XYChart.ts:614
t.update	@	System.ts:157
(anonymous)	@	System.ts:405
requestAnimationFrame (async)		
u	@	AsyncPending.ts:40
t.requestFrame	@	System.ts:404
e.invalidate	@	Sprite.ts:1165
e.invalidate	@	Container.ts:490
set	@	ValueAxis.ts:1626
```

```
Uncaught Error: EventDispatcher is disposed
    at t._on (EventDispatcher.ts:369)
    at t.on (EventDispatcher.ts:444)
    at e.setDataItem (Label.ts:1719)
    at e.set [as dataItem] (Sprite.ts:4203)
    at Container.ts:1852
    at Iterator.ts:100
    at Iterator.ts:45
    at Object.Wglt.e.each (Iterator.ts:99)
    at e.setDataItem (Container.ts:1851)
    at e.set [as dataItem] (Sprite.ts:4203)
t._on	@	EventDispatcher.ts:369
t.on	@	EventDispatcher.ts:444
e.setDataItem	@	Label.ts:1719
set	@	Sprite.ts:4203
(anonymous)	@	Container.ts:1852
(anonymous)	@	Iterator.ts:100
(anonymous)	@	Iterator.ts:45
Wglt.e.each	@	Iterator.ts:99
e.setDataItem	@	Container.ts:1851
set	@	Sprite.ts:4203
e.showTooltip	@	Sprite.ts:8804
e.handleOver	@	Sprite.ts:5525
(anonymous)	@	EventDispatcher.ts:444
(anonymous)	@	EventDispatcher.ts:324
e.each	@	Array.ts:95
t._eachListener	@	EventDispatcher.ts:294
t.dispatchImmediately	@	EventDispatcher.ts:322
e._dispatchSpriteEvent	@	SpriteEvents.ts:53
(anonymous)	@	EventDispatcher.ts:444
(anonymous)	@	EventDispatcher.ts:324
e.each	@	Array.ts:95
t._eachListener	@	EventDispatcher.ts:294
t.dispatchImmediately	@	EventDispatcher.ts:322
e.handleOver	@	Interaction.ts:1456
e.handlePointerOver	@	Interaction.ts:1221
(anonymous)	@	Interaction.ts:539
```

**Comment by martynasma** [comment]
Have you tried enabling auto-dispose of charts?

https://www.amcharts.com/docs/v4/tutorials/chart-was-not-disposed/#Auto_disposing_charts

**Comment by lticadler** [comment]
Didn't know this flag. Thank you.
I'll add this to our code and hopefully the error is gone.

I will keep an eye on it while testing.

**Comment by lticadler** [comment]
Unfortunately our quality control still got this error after adding the global auto dispose flag (and new amcharts version 4.10.10).
![image](https://user-images.githubusercontent.com/58069494/98341406-10008580-200f-11eb-960b-d89a8d31ab91.png)

**Comment by martynasma** [comment]
Would you be able to provide a repro case which we can run ant test?

**Comment by lticadler** [comment]
I'll give it a try...
At the moment I got the error after 10 times changing the date.

**Comment by lticadler** [comment]
Not exactly the same error stack but I was able to create a pen to reproduce a disposer nullpointer every time.
Happens in animated charts with legend if chart data is updated and the date axis shrinks and during that mouse hovers over the "vanishing" part of the chart.

Error happens only in Chrome devtools console - not in the code pen console! (In Firefox the same.)

https://codepen.io/chrisadler/pen/yLJRdZj?editors=1010

![image](https://user-images.githubusercontent.com/58069494/98651515-a6e47f00-233a-11eb-9a14-f044fc7144e1.png)

Stack:
Base.ts:1323 Uncaught TypeError: Cannot read property 'push' of null
```
get | @ | Base.ts:1323
-- | -- | --
  | e.setDataItem | @ | Label.ts:1719
  | set | @ | Sprite.ts:4207
  | e.updateLegendValue | @ | Series.ts:1375
  | e.updateLegendValue | @ | LineSeries.ts:397
  | e.showTooltipAtDataItem | @ | XYSeries.ts:2010
  | (anonymous) | @ | DateAxis.ts:2635
  | e.each | @ | Array.ts:95
  | t.each | @ | List.ts:915
  | e.showTooltipAtPosition | @ | DateAxis.ts:2632
  | (anonymous) | @ | XYChart.ts:1538
  | (anonymous) | @ | Iterator.ts:100
  | (anonymous) | @ | Iterator.ts:45
  | Wglt.e.each | @ | Iterator.ts:99
  | e.showAxisTooltip | @ | XYChart.ts:1535
  | e.handleCursorPositionChange | @ | XYChart.ts:1227
  | (anonymous) | @ | EventDispatcher.ts:444
  | (anonymous) | @ | EventDispatcher.ts:346
  | e.d | @ | AsyncPending.ts:162
  | t.update | @ | System.ts:351
  | (anonymous) | @ | System.ts:405
  | requestAnimationFrame (async) |   |  
  | u | @ | AsyncPending.ts:40
  | t.requestFrame | @ | System.ts:404
  | e.invalidateRawData | @ | Component.ts:1072
```

**Comment by lticadler** [comment]
Any new details about this?

**Comment by martynasma** [comment]
Fixed in [4.10.11](https://github.com/amcharts/amcharts4/releases/tag/4.10.11).

## [4.10.11] - 2020-11-13

### Added
- New setting `reverseGeodata` (default: `false`) on `MapChart` and `MapSeries`. amCharts requires polygon coordinates to be in clockwise order. Some map data have them in counter-clockwise. If your custom map appears garbled, try setting this to `true`.
- New global option: `am4core.options.pixelPerfectPrecision` (default: `0`). Precision (number of decimal points) to be used when rounding point x/y for elements that have `pixelPerfect = true` set.

### Fixed
- `XYChartScrollbar` was not inheriting `inputDateFormat` from the chart properly.
- Clicking on a chart on mobile devices could sometimes make them document scroll a bit.
- Accessibility: pressing ENTER while there is a focused element with `showTooltipOn = "hit"` will not display its tooltip.
- RangeSelector (plugin) was not properly removing its elements after `dispose()` it will now auto-dispose when related axis is disposed.
- In some cases a console error could be displayed when using `XYCursor` over chart when data was being updated.
- Very long labels with `wrap = true` may have been wrapping incorrectly in some cases.
- JSON config: `rgb(...)` and `rgba(...)` syntax was not recongnised in `ColorSet`'s `baseColor`.

[Full change log](https://github.com/amcharts/amcharts4/blob/master/dist/es2015/CHANGELOG.md).

Make sure you clear your browser cache after upgrading. And feel free to contact us again if you are still experiencing this issue.

**Comment by enrialea** [comment]
Hi,

I'm using amCharts 4.10.20 in React

I got the `EventDispatcher is disposed` error, these are the steps:

1. I filter series.values by name to get the correct instance to remove
2. I create a new Serie `newSerie = chartRef.current.series.push(new am4charts.StepLineSeries());`
3. I copy all the properties from old serie like this `am4core.object.copyAllProperties(oldSerie, newSerie);`
4. I remove the old serie `chartRef.current.series.removeIndex(serieToRemove);`

If I only remove a serie without creating new one and copying properties, I got no errors.

This is the console.log @ Series.js:1275 

`Error: EventDispatcher is disposed
    at SpriteEventDispatcher.EventDispatcher.isEnabled (EventDispatcher.js:212)
    at StepLineSeries.BaseObjectEvents.dispatchImmediately (Base.js:1221)
    at StepLineSeries.Component.validateData (Component.js:820)
    at StepLineSeries.XYSeries.validateData (XYSeries.js:707)
    at System.update (System.js:142)
    at System.js:353`
